### PR TITLE
Java docs common

### PR DESCRIPTION
**Pull request by robwrobel**
Added Java docs for a common module.

### CODE REVIEW

File: common/src/main/java/battleships/communication/ClientHandler.java [hunk 1]
```diff
@@ -2,33 +2,35 @@
...
-        .withObjectMapper()
-        .build());
+    this.jsonMarshaller = JsonMarshaller.newInstance();
+    this.jsonUnmarshaller = JsonUnmarshaller.newInstance();
   }
 
+  /**
+   * It first converts message to the string and then send it to receiver.
+   *
+   * @param message Messageable object to be converted to String and then sent by message sender.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/ClientHandler.java]: it's converted to JSON format not to String, JSON uses text representation

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java [hunk 2]
```diff
@@ -2,26 +2,47 @@
...
 public class ClientHandlerBuilder {
   private MessageReceiver messageReceiver;
   private MessageSender messageSender;
   private Socket socket;
 
+  /**
+   * This method should be used as first in ClientHandlerBuilder chain.
+   *
+   * @param socket Socket used by MessageReceiverBuilder and MessageSenderBuilder.
+   * @return this.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: consider change:
return this
to
return instance of ClientHandlerBuilder

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java [hunk 3]
```diff
@@ -2,26 +2,47 @@
...
+   */
   public ClientHandlerBuilder setSocket(Socket socket) {
     this.socket = socket;
     return this;
   }
 
+  /**
+   * Add message receiver to the builder.
+   *
+   * @return this.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: consider change:
return this
to
return instance of ClientHandlerBuilder

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java [hunk 4]
```diff
@@ -2,26 +2,47 @@
...
+   */
   public ClientHandlerBuilder addMessageReceiver() {
     messageReceiver = new MessageReceiverBuilder().addSocket(socket).build();
     return this;
   }
 
+  /**
+   * Add message sender to the builder.
+   * @return this.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: consider change:
return this
to
return instance of ClientHandlerBuilder

File: common/src/main/java/battleships/communication/MessageReceiver.java [hunk 5]
```diff
@@ -6,6 +6,9 @@
 import java.io.IOException;
 import java.io.ObjectInputStream;
 
+/**
+ * This class is used to deserialize given ObjectInputStream to receive String object.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/MessageReceiver.java]: it is text representation of JSON representing Messagable, not just String

File: common/src/main/java/battleships/communication/MessageReceiverBuilder.java [hunk 6]
```diff
@@ -7,6 +7,9 @@
 import java.io.ObjectInputStream;
 import java.net.Socket;
 
+/**
+ * This class is for building MessageReceiver object, based on socket input stream.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/MessageReceiverBuilder.java]: For me this part is unnecessary: "based on socket input stream."
Moreover I will be interested in some explanation what is that class responsible for.

File: common/src/main/java/battleships/communication/MessageSender.java [hunk 7]
```diff
@@ -5,6 +5,9 @@
 import java.io.IOException;
 import java.io.ObjectOutputStream;
 
+/**
+ * This class is used to serialize String message to the given ObjectOutputStream.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/MessageSender.java]: Isn't is just writing String message to ObjectOutputStream?
where is serialization in here happening?

File: common/src/main/java/battleships/communication/MessageSenderBuilder.java [hunk 8]
```diff
@@ -7,6 +7,9 @@
 import java.io.OutputStream;
 import java.net.Socket;
 
+/**
+ * This class is for building MessageSender object, based on socket output stream.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/MessageSenderBuilder.java]: What that class is for? It does not explain that

File: common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java [hunk 9]
```diff
@@ -5,14 +5,23 @@
 import battleships.utils.BattleshipUtils;
 import com.fasterxml.jackson.core.JsonProcessingException;
 
+/**
+ * This class is a Json implementation of Marshaller. Its main responsibility is to turn a Messageable
+ * object into its String representation using MessageableMapper wrapper class.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java]: its not just String representation, it's String which is JSON representation (JSON is text format)

Comment (igasm) [reply common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java]: I think that you can name it  Json String, as you done it below (in JsonUnmarshaller class JavaDoc)

File: common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java [hunk 10]
```diff
@@ -5,14 +5,23 @@
...
   private final MessageableMapper messageableMapper;
 
-  public JsonMarshaller(MessageableMapper messageableMapper) {
+  private JsonMarshaller(MessageableMapper messageableMapper) {
     this.messageableMapper = messageableMapper;
   }
 
+  /**
+   * @param messageable - Object of type Messageable to be converted to String
+   * @return - String representation of the Messageable object.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java]: it's return JSON String representation of the Messageable object.

File: common/src/main/java/battleships/communication/messages/Salvo.java [hunk 11]
```diff
@@ -8,19 +8,34 @@
...
+  /**
+   * @param positions vararg of positions where salvo is shoted
+   * @return new instance of Salvo object.
+   */
   public static Salvo createForPositions(Integer... positions) {
     return new Salvo(Arrays.stream(positions).collect(Collectors.toList()));
   }
 
+  /**
+   * This is a constructor for Salvo. It is used by Json as a property based creator.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/messages/Salvo.java]: It is used by Jackson library, not Json itself.

File: common/src/main/java/battleships/communication/messages/WelcomeMessage.java [hunk 12]
```diff
@@ -4,10 +4,18 @@
...
 
+/**
+ * This class represents welcome message that is send from a server to clients when they connect to the server.
+ */
 public class WelcomeMessage implements Messageable {
 
   private final String body;
 
+  /**
+   * This is a constructor for WelcomeMessage. It is used by Json as a property based creator.
```
Comment (igasm) [inline common/src/main/java/battleships/communication/messages/WelcomeMessage.java]: It is used by Jackson library, not Json itself.

File: common/src/main/java/battleships/communication/ClientHandler.java (hunk 1 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/ClientHandler.java]: done

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java (hunk 2 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: done

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java (hunk 3 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: done

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java (hunk 4 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: done

File: common/src/main/java/battleships/communication/MessageReceiver.java (hunk 5 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/MessageReceiver.java]: done

File: common/src/main/java/battleships/communication/MessageReceiverBuilder.java (hunk 6 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/MessageReceiverBuilder.java]: done

File: common/src/main/java/battleships/communication/MessageSender.java (hunk 7 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/MessageSender.java]: done

File: common/src/main/java/battleships/communication/MessageSenderBuilder.java (hunk 8 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/MessageSenderBuilder.java]: done

File: common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java (hunk 9 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java]: done

File: common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java (hunk 10 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java]: done

File: common/src/main/java/battleships/communication/messages/WelcomeMessage.java (hunk 12 above)
Comment (robwrobel) [reply common/src/main/java/battleships/communication/messages/WelcomeMessage.java]: done

**Pull request merged**